### PR TITLE
Feat/parser

### PR DESCRIPTION
**Pull request by Felixsta123**
Merging the simple parser to dev

### CODE REVIEW

File: server/src/parser.cpp [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+/**
+ * @brief Extract the value from a pair
+ * 
+ * @param pair 
+ * @return std::string 
+ */
+std::string Parser::extractValue(const std::string& pair) {
+    size_t pos = pair.find(':');
+    return pair.substr(pos + 1);
+}
```
Comment (PaulThblt) [inline server/src/parser.cpp:55]: Newline pls
```suggestion
}

```

File: server/includes/parser.hpp [hunk 2]
```diff
@@ -0,0 +1,52 @@
+/**
+ * @file parser.hpp
+ * @brief 
+ * @date 26-12-2023
+ */
+
+// message example:x:0-y:10-hp:40-ip:127.0.0.1:13151
```
Comment (PaulThblt) [inline server/includes/parser.hpp:7]: It would be cool to have a better documented norm for the messages (maybe in the README? or a separate document)

**Review by PaulThblt — CHANGES_REQUESTED** [review]
You can just accept the suggested code snippet

File: server/includes/parser.hpp (hunk 2 above)
Comment (PaulThblt) [reply server/includes/parser.hpp:7]: I think it would be useful for the review tomorrow

Comment (Felixsta123) [reply server/includes/parser.hpp:7]: I agree, on it

**Review by PaulThblt — APPROVED** [review]

**Pull request merged**